apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/Rates/Filter.jsx

```diff
@@ -0,0 +1,80 @@
+// @flow
+import React, { useState } from "react";
+import { Trans } from "react-i18next";
+import Box from "~/renderer/components/Box";
+import styled from "styled-components";
+import { FILTER } from "../utils";
+import Button from "~/renderer/components/Button";
+import { Icons } from "@ledgerhq/react-ui";
+
+type Props = {
+  onClick: () => void,
+};
+
+const Container: ThemedComponent<{}> = styled(Box).attrs(() => ({}))`
+  flex-direction: row;
+  margin: 10px 0;
+`;
+
+export const Btn: ThemedComponent<{}> = styled(Button).attrs((p) => {
+  return {
+    key: p.key,
+    primary: p.selected,
+    outline: p.selected,
+    onClick: () => p.updateFilter(p.type),
+    iconPosition: "left",
+  };
+})`
+  border-radius: 500px;
+  margin-right: 12px;
+  padding: 0 12px;
+  height: 32px;
+  color: ${(p) => (p.selected ? p.theme.colors.neutral.c00 : p.theme.colors.neutral.c100)};
+  background-color: ${(p) =>
+    p.selected ? p.theme.colors.primary.c80 : p.theme.colors.neutral.c30};
+  &:active,
+  &:hover {
+    color: ${(p) => (p.selected ? p.theme.colors.primary.c80 : p.theme.colors.neutral.c100)};
+  }
+`;
+
+export default function ProviderRate({ onClick }: Props) {
+  const [filter, setFilter] = useState([]);
+
+  const updateFilter = (type) => {
+    let newFilter = [];
+    if (filter.includes(type)) {
+      newFilter = filter.filter((e) => e !== type);
+      setFilter(newFilter);
+    } else {
+      setFilter([type, ...filter]);
+      newFilter = [type, ...filter];
+    }
+    if ([FILTER.centralised, FILTER.decentralised].every((elem) => newFilter.includes(elem))) {
+      newFilter = newFilter.filter((f) => ![FILTER.centralised, FILTER.decentralised].includes(f));
+    }
+    if ([FILTER.fixed, FILTER.float].every((elem) => newFilter.includes(elem))) {
+      newFilter = newFilter.filter((f) => ![FILTER.fixed, FILTER.float].includes(f));
+    }
```

Comment:
i'm a bit confused why you need both the "newFilter" internal let variable which diverges from the `filter` **state**. 

the code also isn't so easy to understand so maybe worth having more documentation (or maybe there is a simpler way to achieve what's done?)

Reply:
I have had some comments to clarify.

If centralised & decentralised filters are active, these 2 filters will not be applied
If fixed & float filters are active, these 2 filters will not be applied

(for that reason the status of the component not represent exactly the filter itself)